evaluator/evaluator.py

```diff
@@ -216,36 +244,49 @@ async def evaluate_vmaas(system_platform, acc_num, org_id, conn):
     LOGGER.info("Evaluated vulnerabilities for inventory_id: %s (%s playbook, %s manual)", inventory_id,
                 len(reported_playbook_cves), len(reported_manually_fixable_cves))
     system_cves_map = await _load_cves_for_inventory_id(rh_account_id, system_id, conn)
-    unprocessed_cves = set(system_cves_map.keys())
-    new_cves = set()
+    unprocessed_cves = dict.fromkeys(system_cves_map.keys(), ErrataHasPlaybook(None, False))
+    new_cves = dict()
     mitigated_cves = set()
-    unmitigated_cves = set()
+    unmitigated_cves = dict()
     system_cves = []
     mit_sys_vulns = []
     unmit_sys_vulns = []
 
+    # TODO: DELETE LATER, we need to populate advisories column for current CVEs
+    cves_needed_update = []
+
     for cve in reported_playbook_cves:
-        system_cves.append(cve)
-        if cve in system_cves_map:
-            unprocessed_cves.discard(cve)
-            if system_cves_map[cve]['when_mitigated']:
+        system_cves.append(cve.name)
+        if cve.name in system_cves_map:
+            unprocessed_cves.pop(cve.name)
+            if system_cves_map[cve.name]['when_mitigated']:
                 # it was mitigated, now its not
-                unmitigated_cves.add((cve, True))
-                unmit_sys_vulns.append((system_cves_map[cve]["sv_id"], system_cves_map[cve]["cve_id"]))
+                unmitigated_cves[cve.name] = ErrataHasPlaybook(cve.errata, True)
+                unmit_sys_vulns.append((system_cves_map[cve.name]["sv_id"], system_cves_map[cve.name]["cve_id"]))
+            elif CFG.update_cve_advisories:
```

Comment:
Please add conditon to do the update only if advisories is not yet filled or is different than value in DB.